libraries/ui/src/main/java/androidx/media3/ui/PlayerControlView.java

```diff
@@ -1574,14 +1574,25 @@ private void seekToTimeBarPosition(Player player, long positionMs) {
   }
 
   private void onFullScreenButtonClicked(View v) {
-    if (onFullScreenModeChangedListener == null) {
+    updateIsFullScreen(!isFullScreen);
+  }
+
+  /**
+   * Updates whether the controller is in full screen to its icon
+   * and reports it to to the listener.
+   * @param isFullScreen If the view is in full screen.
+   */
+  public void updateIsFullScreen(boolean isFullScreen) {
+    boolean fullScreenModeChanged = this.isFullScreen != isFullScreen;
+
+    if(fullScreenModeChanged && onFullScreenModeChangedListener == null) {
```

Comment:
I think this line can be simplified to `if(onFullScreenModeChangedListener == null)`.

No matter whether the mode changed or not, if the listener is not there, we don't do anything

Reply:
Don't we want to update the internal state whether or not we have a listener attached?

Consider:

```java
playerView.setFullscreenButtonState(true);
playerView.setFullscreenButtonClickListener(isFullScreen -> ...);
```

Registering the listener means the button is shown, but the "enter full screen" button will be shown even though we've been told the content is full screen above, and so the "exit full screen" **should** be shown.

Reply:
Sorry, yes, you are right - got confused with the previous impl of:

```
  private void onFullScreenButtonClicked(View v) {
    if (onFullScreenModeChangedListener == null) {  <-- can't be true, the button is not visible if the listener==null
      return;
    }
   ...
```

So I guess this line could be changed to `if (!fullScreenModeChanged)` to exit early?